src/legacy/ui/public/chrome/api/sub_url_hooks_utils.test.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { areDecodedHashesEqual } from './sub_url_hooks_utils';
+
+test('false for different hashes', () => {
+  const url1 = `https://localhost/kibana/#/dashboard/id`;
+  const url2 = `https://localhost/kibana/#/dashboard/DIFFERENT`;
+  expect(areDecodedHashesEqual(url1, url2)).toBeFalsy();
+});
```

Comment:
Would add a `false for same hashes` test as if I understand correctly `areDecodedHashesEqual(urlWithHash, urlWithHash)` returns false.